nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AnnotationServiceImpl.java

```diff
@@ -89,7 +89,7 @@ public Annotation getAnnotation(CoreSession session, String annotationId) {
         }
         DocumentModel annotatedDoc = session.getDocument(docRef);
         return Framework.getService(CommentManager.class)
-                        .getComments(annotatedDoc)
+                        .getComments(session, annotatedDoc)
```

Comment:
Why not using page provider introduced in 10.2 ? This will also allow us to not have `getComments(CoreSession, DocumentModel).

Reply:
We will use the migration state to switch between the page provider and this method with https://jira.nuxeo.com/browse/NXP-25429